# Heroes Of The Land - Game Data Analysis

* Of the 1163 active players, the vast majority are male (84%). There also exists, a smaller, but notable proportion of female players (14%).

* Our peak age demographic falls between 20-24 (44.8%) with secondary groups falling between 15-19 (18.60%) and 25-29 (13.4%).  
-----

In [4]:
# Dependencies and Setup
import pandas as pd
import numpy as np

# File to Load (Remember to Change These)
file_to_load = "./resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)


## Player Count

## The total number of players




In [5]:
# set up purchase data into a data frame
df_organized = pd.DataFrame(purchase_data)
df_organized.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [6]:
# calculate total number of unique players
total_players = len(df_organized.SN.value_counts())
total_players
print(f"The total number of players are {total_players}.")

The total number of players are 576.


## Purchasing Analysis (Total)

* Calculations of number of unique items, average price, etc.

* Created a summary dataframe of the results


In [7]:
# Run basic calculations to obtain number of unique items, average price, etc.

In [8]:
# total unique items
total_unique_items = len(df_organized['Item ID'].value_counts())
total_unique_items

183

In [9]:
# total unique item names
unique_item_name = len(df_organized['Item Name'].value_counts())
unique_item_name

179

In [10]:
# calculate average price
average_price = round(df_organized.Price.mean(), 2)
average_price

3.05

In [11]:
# Calculated total revenue by summing up all prices that were sold to players
total_revenue = round(df_organized.Price.sum(), 2)
total_revenue

2379.77

In [12]:
# total unique_SN
total_unique_SN = len(df_organized['SN'].value_counts())
total_unique_SN

576

In [13]:
# average purchase per player
avg_purchase_PP = round(total_revenue / total_unique_SN, 2)
avg_purchase_PP

4.13

In [14]:
# total unique purchase ID's. later realized these were transaction numbers
total_unique_purchID = len(df_organized['Purchase ID'].value_counts())
total_unique_purchID

780

In [15]:
# average age
average_age = round(df_organized.Age.mean())
average_age

23

In [16]:
#df_organized.head()

### High Level Summary Of Data Related to Audience, Sales, and Age

In [17]:
# Create a summary data frame to hold the results and display summary
purch_analysis_sumdf = pd.DataFrame({"Total # of Unique SN's": [total_unique_SN],
                                     "Total # of Purchases": [total_unique_purchID],
                                     "Total Unique Items Offered": [total_unique_items],
                                     "Total Revenue in $": [total_revenue],
                                     "Average Purchase Per Player in $": [avg_purchase_PP],
                                     "Average Price in $": [average_price],
                                     "Average Age": [average_age],
                                    })                  
purch_analysis_sumdf

Total # of Unique SN's  Total # of Purchases  Total Unique Items Offered  \
0                     576                   780                         183   

   Total Revenue in $  Average Purchase Per Player in $  Average Price in $  \
0             2379.77                              4.13                3.05   

   Average Age  
0           23

## Gender Demographics

* Percentage and Count of Male Players


* Percentage and Count of Female Players


* Percentage and Count of Other / Non-Disclosed




In [18]:
# located all male players in purchase_data
male_df = df_organized.loc[df_organized["Gender"] == "Male" , :]
male_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [19]:
# grouped all male players by their unique SN
grouped_male_df = male_df.groupby(['SN'])
#print(grouped_male_df)

# total male players
total_male_players = len(grouped_male_df)
total_male_players

484

In [20]:
# located all female players in the purchase data set
female_df = df_organized.loc[df_organized["Gender"] == "Female" , :]
female_df.head()

Purchase ID         SN  Age  Gender  Item ID  \
15           15  Lisassa64   21  Female       98   
18           18  Reunasu60   22  Female       82   
38           38   Reulae52   10  Female      116   
41           41  Assosia88   20  Female        7   
55           55  Phaelap26   25  Female       84   

                       Item Name  Price  
15   Deadline, Voice Of Subtlety   2.89  
18                       Nirvana   4.90  
38       Renewed Skeletal Katana   4.18  
41  Thorn, Satchel of Dark Souls   1.33  
55                    Arcane Gem   3.79

In [21]:
# group the female player data by their SN's
grouped_female_df = female_df.groupby(['SN'])
#print(grouped_female_df)

#calculate the number of female players
total_female_players = len(grouped_female_df)
total_female_players

81

In [22]:
# located all other players
other_df = df_organized.loc[df_organized["Gender"] == "Other / Non-Disclosed" , :]
other_df.head()

Purchase ID            SN  Age                 Gender  Item ID  \
9              9   Chanosian48   35  Other / Non-Disclosed      136   
22            22  Siarithria38   38  Other / Non-Disclosed       24   
82            82    Haerithp41   16  Other / Non-Disclosed      160   
111          111      Sundim98   21  Other / Non-Disclosed       41   
228          228   Jiskirran77   20  Other / Non-Disclosed       80   

                        Item Name  Price  
9    Ghastly Adamantite Protector   3.58  
22                  Warped Fetish   3.81  
82                     Azurewrath   4.40  
111                         Orbit   4.75  
228                     Dreamsong   3.39

In [23]:
# group the other player data by their SN's

grouped_other_df = other_df.groupby(['SN'])
#print(grouped_other_df)

#calculated all other oplayers
total_other_players = len(grouped_other_df)
total_other_players

11

### Calculate percentages of players by gender

In [24]:
male_percentage = round(total_male_players / total_players, 4)*100
male_percentage

84.03

In [25]:

female_percentage = round(total_female_players / total_players, 4)*100
female_percentage

14.06

In [26]:

other_percentage = round(total_other_players / total_players, 4)*100
other_percentage

1.91

### Males make up about 5/6 of the total gaming audience for this game.

In [27]:
player_summary_df = pd.DataFrame({"Total Count": [total_male_players, total_female_players,
                                                  total_other_players],
                                 "Percentage of Players": [male_percentage, female_percentage,
                                                           other_percentage]},
                                 index=["Male", "Female", "Other / Non-Disclosed"])            
player_summary_df

Total Count  Percentage of Players
Male                           484                  84.03
Female                          81                  14.06
Other / Non-Disclosed           11                   1.91


## Purchasing Analysis (Gender)

* Ran calculations for purchase count, avg. purchase price, avg. purchase total per person etc. by gender

* Created a summary dataframe to display the results


Total Purchase Count Calculations:

In [28]:
female_df = df_organized.loc[df_organized["Gender"] == "Female" , :]

total_purchase_countF = len(female_df)

total_purchase_countF

113

In [29]:
male_df = df_organized.loc[df_organized["Gender"] == "Male" , :]

total_purchase_countM = len(male_df)

total_purchase_countM

652

In [30]:
other_df = df_organized.loc[df_organized["Gender"] == "Other / Non-Disclosed" , :]

total_purchase_countO = len(other_df)

total_purchase_countO

15

Average Purchase Price Calculations:

In [31]:
female_df = df_organized.loc[df_organized["Gender"] == "Female" , :]

avg_purch_priceF = round(female_df['Price'].mean(), 2)

avg_purch_priceF

#print(str(avg_purch_priceF) +"0")


3.2

In [32]:
male_df = df_organized.loc[df_organized["Gender"] == "Male" , :]

avg_purch_priceM = round(male_df['Price'].mean(), 2)

avg_purch_priceM


3.02

In [33]:
other_df = df_organized.loc[df_organized["Gender"] == "Other / Non-Disclosed" , :]

avg_purch_priceO = round(other_df['Price'].mean(), 2)

avg_purch_priceO


3.35

Total Purchase Price Calculations:

In [34]:
female_df = df_organized.loc[df_organized["Gender"] == "Female" , :]

sum_purch_priceF = round(female_df['Price'].sum(), 2)

sum_purch_priceF

361.94

In [35]:
male_df = df_organized.loc[df_organized["Gender"] == "Male" , :]

sum_purch_priceM = round(male_df['Price'].sum(), 2)

sum_purch_priceM

1967.64

In [36]:
other_df = df_organized.loc[df_organized["Gender"] == "Other / Non-Disclosed" , :]

sum_purch_priceO = round(other_df['Price'].sum(), 2)

sum_purch_priceO

50.19

Average Purchase Per Person Calculations:

In [37]:
avg_purch_perF = round(sum_purch_priceF / total_female_players, 2)
avg_purch_perF

4.47

In [38]:
avg_purch_perM = round(sum_purch_priceM / total_male_players, 2)
avg_purch_perM

4.07

In [39]:
avg_purch_perO = round(sum_purch_priceO / total_other_players, 2)
avg_purch_perO

4.56

## Female gamers may spend on average slightly more than their male counter parts (3.20 (F) vs 3.02 (M)), but the purchase activity suggests that the male population is significantly larger and their purchasing volume was approximately six times greater than their female and other / non-disclosed counterparts.  

In [42]:
# Created a summary maual dataframe for results

player_summary_df = pd.DataFrame({"Purchase Count": [total_purchase_countF, total_purchase_countM,
                                                     total_purchase_countO],
                                  "Average Purchase Price": [avg_purch_priceF, avg_purch_priceM,
                                                             avg_purch_priceO],
                                  "Total Purchase Value": [sum_purch_priceF, sum_purch_priceM,
                                                           sum_purch_priceO],
                                  "Avg Total Purchase per Person": [avg_purch_perF, avg_purch_perM,
                                                                    avg_purch_perO],
                                  "Gender": ["Female", "Male", "Other / Non-Disclosed"]
                                 })

# Format values in the summary table
player_summary_df["Average Purchase Price"] = player_summary_df["Average Purchase Price"].map("${:.2f}".format)
player_summary_df["Total Purchase Value"] = player_summary_df["Total Purchase Value"].map("${:.2f}".format)
player_summary_df["Avg Total Purchase per Person"] = player_summary_df["Avg Total Purchase per Person"].map("${:.2f}".format)

# Note always need to state the object, then restate it again then the method to make changes and save it to the same
# object like how you did it down below
player_summary_df = player_summary_df.set_index("Gender")

player_summary_df

Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Avg Total Purchase per Person  
Gender                                                                    
Female                             $361.94                         $4.47  
Male                              $1967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

## Age Demographics

* Established bins for ages

* Categorized the existing players using the age bins. Hint: use pd.cut()

* Calculated the numbers and percentages by age group

* Created a summary dataframe to display results

* Rounded the percentage column to two decimal points

* Displayed Age Demographics Table


In [39]:
# Understand the range of data
print(df_organized["Age"].max())
print(df_organized["Age"].median())
print(df_organized["Age"].min())

45
22.0
7


In [40]:
df_organized.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [41]:
# Created bins in which to place values based upon ages
# for age_bins, I put 9 because the ask stated the bin should be "<10" and when I put 10, all ages of 10 were included

age_bins = [0, 9, 14, 19, 24, 29, 34, 39, 100]
# Create labels for these bins
bin_labels = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", 
               "35-39", "40+"]

In [42]:
# Slice the data and place it into bins
pd.cut(df_organized["Age"], age_bins, labels=bin_labels).head()

0    20-24
1      40+
2    20-24
3    20-24
4    20-24
Name: Age, dtype: category
Categories (8, object): [<10 < 10-14 < 15-19 < 20-24 < 25-29 < 30-34 < 35-39 < 40+]

In [43]:
# Place the data series into a new column inside of the DataFrame
df_organized["Age Group"] = pd.cut(df_organized["Age"], age_bins, labels=bin_labels)

sn_group_df = df_organized.groupby(["SN"])

age_group_valuects = sn_group_df["Age Group"].value_counts()

### The age group of  15-24 MAKES UP ~74% (20-24: 45% AND 15-19: 19%). Marketing efforts should be tailored towards this audience.

In [44]:
# put value counts into data frame
age_group_valuects = pd.DataFrame(age_group_valuects)

age_group_valuects = age_group_valuects.rename(columns={"Age Group":"Counts"})

age_group_valuects = age_group_valuects.reset_index(level=["Age Group"])
age_group_valuects = pd.DataFrame(age_group_valuects["Age Group"].value_counts())
age_group_valuects = age_group_valuects.rename(columns={"Age Group":"Total Count"})
age_group_valuects = age_group_valuects.reindex(["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"])

#adding a column to the dataframe age_group_counts_df called "Percentage of Players"
# and will figure out the formula for you.

sum_age_valuects = age_group_valuects["Total Count"].sum()

age_group_valuects["Percentage of Players"] = (age_group_valuects["Total Count"]/sum_age_valuects)

#Formatting the value in the summary table
age_group_valuects['Percentage of Players'] = pd.to_numeric(age_group_valuects['Percentage of Players'])
age_group_valuects['Percentage of Players'] = age_group_valuects['Percentage of Players'].map("{:.2%}".format)

age_group_valuects

Total Count Percentage of Players
<10             17                 2.95%
10-14           22                 3.82%
15-19          107                18.58%
20-24          258                44.79%
25-29           77                13.37%
30-34           52                 9.03%
35-39           31                 5.38%
40+             12                 2.08%

## Purchasing Analysis (Age)

* Binned the purchase_data data frame by age

* Ran basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below

* Created a summary dataframe to display results


In [47]:
age_group = df_organized.groupby("Age Group")

total_purch_age = age_group["Purchase ID"].count()
print(total_purch_age)


Age Group
<10       23
10-14     28
15-19    136
20-24    365
25-29    101
30-34     73
35-39     41
40+       13
Name: Purchase ID, dtype: int64


In [48]:
total_age_percentage = round((age_group["SN"].count() / total_unique_SN)*100, 2)
print(total_age_percentage)

Age Group
<10       3.99
10-14     4.86
15-19    23.61
20-24    63.37
25-29    17.53
30-34    12.67
35-39     7.12
40+       2.26
Name: SN, dtype: float64


In [49]:
avg_purchPrice_age = round((age_group["Price"].sum() / total_purch_age), 2)
print(avg_purchPrice_age)

Age Group
<10      3.35
10-14    2.96
15-19    3.04
20-24    3.05
25-29    2.90
30-34    2.93
35-39    3.60
40+      2.94
dtype: float64


In [50]:
total_purch_value = age_group["Price"].sum()
print(total_purch_value)

Age Group
<10        77.13
10-14      82.78
15-19     412.89
20-24    1114.06
25-29     293.00
30-34     214.00
35-39     147.67
40+        38.24
Name: Price, dtype: float64


In [51]:

avg_purchPrice_person = round((total_purch_value / total_purch_age), 2)
print(avg_purchPrice_person)

Age Group
<10      3.35
10-14    2.96
15-19    3.04
20-24    3.05
25-29    2.90
30-34    2.93
35-39    3.60
40+      2.94
dtype: float64


In [52]:
purch_perSN = pd.DataFrame(age_group["SN"].unique())
total_purch_perSN = purch_perSN["SN"].str.len()

avg_purchPrice_person = round((total_purch_value / total_purch_perSN), 2)
print(avg_purchPrice_person)

Age Group
<10      4.54
10-14    3.76
15-19    3.86
20-24    4.32
25-29    3.81
30-34    4.12
35-39    4.76
40+      3.19
dtype: float64


### The most age group that purchased the most was 20-24 with a total purchase of 1,114.06, and a volume of 365; this is followed by 15-19 with 136 purchases and a total spend of 412.89. 

In [53]:
#Summary Dataframe
purch_age_summdf = pd.DataFrame({"Purchase Count": [total_purch_age[1], total_purch_age[2],
                                total_purch_age[3], total_purch_age[4],total_purch_age[5], total_purch_age[7],
                                total_purch_age[7], total_purch_age[0]],
                                 "Average Purchase Price": [avg_purchPrice_age[1], avg_purchPrice_age[2],
                                avg_purchPrice_age[3], avg_purchPrice_age[4],avg_purchPrice_age[5], avg_purchPrice_age[7],
                                avg_purchPrice_age[7], avg_purchPrice_age[0]],
                                 "Total Purchase Value": [total_purch_value[1], total_purch_value[2],
                                total_purch_value[3], total_purch_value[4],total_purch_value[5], total_purch_value[7],
                                total_purch_value[7], total_purch_value[0]],
                                "Average Total Purchase per Person": [avg_purchPrice_person[1], avg_purchPrice_person[2],
                                avg_purchPrice_person[3], avg_purchPrice_person[4],avg_purchPrice_person[5], avg_purchPrice_person[7],
                                avg_purchPrice_person[7], avg_purchPrice_person[0]],
                                "  ": ["10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+", "<10"]})
purch_age_summdf.set_index('  ')

Purchase Count  Average Purchase Price  Total Purchase Value  \
                                                                      
10-14              28                    2.96                 82.78   
15-19             136                    3.04                412.89   
20-24             365                    3.05               1114.06   
25-29             101                    2.90                293.00   
30-34              73                    2.93                214.00   
35-39              13                    2.94                 38.24   
40+                13                    2.94                 38.24   
<10                23                    3.35                 77.13   

       Average Total Purchase per Person  
                                          
10-14                               3.76  
15-19                               3.86  
20-24                               4.32  
25-29                               3.81  
30-34                               4.12  
35-39                               3.19  
40+                                 3.19  
<10                                 4.54

In [54]:
#couldn't get the <10 to be the last row, so it the manual dataframe way above
# age_groups_purchase_df = pd.concat([total_purch_age, avg_purchPrice_age, 
#                                     total_purch_value, avg_purchPrice_person], axis=1)
# age_groups_purchase_df

## Top Spenders

In [59]:
#sn_group_df.head()

In [91]:
# Find the purchase count
purchase_ct = sn_group_df['Purchase ID'].count()
#purchase_ct.head()
# #type(purchase_ct)


In [90]:
#lesson learned is skim thru all data bc
#calculate total purchase value
total_purchval = round(sn_group_df['Price'].sum(), 2)
#total_purchval
# type(total_purchval)

In [60]:
#calculate avg purchase value
avg_purch = round(total_purchval / purchase_ct, 2)
type(avg_purch)

pandas.core.series.Series

In [81]:
# after concatenation, skim thru the whole list
sn_purchases_df = pd.concat([purchase_ct, avg_purch, total_purchval], axis=1)
#sn_purchases_df.head()
#type(sn_purchases_df)

In [82]:
#type(sn_purchases_df)

In [83]:
sn_purchases_df = sn_purchases_df.sort_values(by=["Price"], ascending=False)
#sn_purchases_df.head

In [84]:
sn_purchases_df = sn_purchases_df.rename(columns={"Purchase ID": "Purchase Count", 0: "Average Purchase Price",
                                                 "Price": "Total Purchase Value"})
#sn_purchases_df.head()

In [65]:
# format the table
sn_purchases_df["Average Purchase Price"] = sn_purchases_df["Average Purchase Price"].map("${:.2f}".format)
sn_purchases_df["Total Purchase Value"] = sn_purchases_df["Total Purchase Value"].map("${:.2f}".format)

### Top 5 players: Total Purchase Counts, Average Purchase Price and Total Purchase Value

In [66]:
sn_purchases_df.head()

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

## Most Popular Items

* Retrieved the Item ID, Item Name, and Item Price columns of the most popular items.

* Grouped by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value

* Created a summary data frame to hold the results

* Sorted the purchase count column in descending order

In [70]:
# Retrieve the Item ID, Item Name, and Item Price columns

df_items = df_organized[["Item ID", "Item Name", "Price"]]

In [85]:
# Group by Item ID and Item Name
popular_items_df = df_items.groupby(['Item ID', 'Item Name'])
#type(popular_items_df.head())
#popular_items_df.head()

In [86]:
# calculate count of poplular items purchased and store into data frame
item_purch_ct = popular_items_df.count()
#type(item_purch_ct)

item_purch_ct_df = pd.DataFrame(item_purch_ct)
#(item_purch_ct_df.head())

# rename the column
item_purch_ct_df = item_purch_ct_df.rename(columns={"Price": "Purchase Count"})
#item_purch_ct_df.head()


In [87]:
# calculate total purchase value same way as above
item_total_purchval = popular_items_df.sum()
type(item_total_purchval)

item_total_purchval_df = pd.DataFrame(item_total_purchval)
item_total_purchval_df = item_total_purchval_df.rename(columns={"Price":"Total Purchase Value"})
#item_total_purchval_df.head()

In [88]:
# calculate item price and store into a dataframe

item_price = (item_total_purchval / item_purch_ct)
#type(item_price_df)
item_price_df = pd.DataFrame(item_price)

# rename column
item_price_df = item_price_df.rename(columns={"Price":"Item Price"})
#item_price_df.head()

### Top 5 Most Popular In-app Game Purchases

In [79]:
#concat all three dataframes  by the column axis (axis=1)
item_summary_table_df = pd.concat([item_purch_ct_df, item_price_df, item_total_purchval_df], axis=1)

# sort the data by descending
item_summary_table_df = item_summary_table_df.sort_values(by='Purchase Count', ascending=False)

#format the dataframe columns
item_summary_table_df["Item Price"] = item_summary_table_df["Item Price"].map("${:.2f}".format)
item_summary_table_df["Total Purchase Value"] = item_summary_table_df["Total Purchase Value"].map("${:.2f}".format)

item_summary_table_df.head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
108     Extraction, Quickblade Of Trembling Hands                  9   
82      Nirvana                                                    9   
19      Pursuit, Cudgel of Necromancy                              8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
108     Extraction, Quickblade Of Trembling Hands         $3.53   
82      Nirvana                                           $4.90   
19      Pursuit, Cudgel of Necromancy                     $1.02   

                                                     Total Purchase Value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
108     Extraction, Quickblade Of Trembling Hands                  $31.77  
82      Nirvana                                                    $44.10  
19      Pursuit, Cudgel of Necromancy                               $8.16

In [45]:
#item_summary_table_df.head()

In [77]:
# find most profitable item. had to recreate the concat and call it a new table because was having errors and not
# getting results wanted. perhaps circular referencing on the backend

most_profitable_df = pd.concat([item_purch_ct_df, item_price_df, item_total_purchval_df], axis=1)
# sort values

most_profitable_df = most_profitable_df.sort_values(by='Total Purchase Value', ascending=False)

In [78]:
most_profitable_df.head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
92      Final Critic                                               8   
103     Singed Scalpel                                             8   

                                                      Item Price  \
Item ID Item Name                                                  
178     Oathbreaker, Last Hope of the Breaking Storm        4.23   
82      Nirvana                                             4.90   
145     Fiery Glass Crusader                                4.58   
92      Final Critic                                        4.88   
103     Singed Scalpel                                      4.35   

                                                      Total Purchase Value  
Item ID Item Name                                                           
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76  
82      Nirvana                                                      44.10  
145     Fiery Glass Crusader                                         41.22  
92      Final Critic                                                 39.04  
103     Singed Scalpel                                               34.80

# Data Analysis Observations:

* The game is averaging about 1.35 transaction per unique player. Not all player makes purchases in the game.
* Male players made up 85% of their purchases for a total of 1,967, averaging about 4.07 per player.
* About %75 of  their players are between the ages of 15 to 29, with 45% of them being in the age group of 20 – 24.
* Players in the age group of 20 -24 had the highest purchase count (365) and spent on average of 4.32.
* Conversely, the lowest number of transactions and spend came both age groups of 35 to 39, and 40, where both groups had     13 transactions with an average of 3.20.
* Their top spenders spent between 13.10 to 18.96, with minimum of 3 to max 5 purchase counts.
* Their most profitable item was “Oathbreaker, Last Hope of the Breaking Storm” for a total of 12 purchases with a total     revenue of 50.76.

